apps/loan/loan-products/src/app/shared/services/loan-products.service.ts

```diff
@@ -121,11 +131,11 @@ export class LoanProductsService {
   }
 
   /**
-   * Upsert a subproduct into a loan products subproduct array
-   * @param product
-   * @returns
+   * Upserts a non-credit product into the loan products array. If the product already exists, it updates it; otherwise, it adds it to the array.
+   * @param {LoanProductModels.NonCreditProduct} product - The non-credit product to be upserted.
+   * @returns {void}
    */
-  public subProductUpsert(product: LoanProductModels.SubProduct) {
+  public nonCreditProductUpsert(product: LoanProductModels.NonCreditProduct) {
```

Comment:
is this "upsert" word new? I've always just used `save` to imply insert or update.

Reply:
Yeah I like that